pymeasure/adapters/adapter.py

```diff
@@ -114,6 +119,33 @@ def binary_values(self, command, header_bytes=0, dtype=np.float32):
         raise NameError("Adapter (sub)class has not implemented the "
                         "binary_values method")
 
+    def __getattribute__(self, name):
```

Comment:
```
def __getattribute__(self, name):
        """ Override to acquire the 'connection_lock' lock through the duration of attribute access.
        """
        if not name in ("connection_lock, "connection_lock_initialized") and self.connection_lock_initialized:
            with self.connection_lock:
                ret = super().__getattribute__(name)
        else:
            ret = super().__getattribute__( name)
```

Reply:
I like this version better as well. I've changed it to this.